webview-ui/src/components/settings/ToolSettings.tsx

```diff
@@ -0,0 +1,170 @@
+import { VSCodeCheckbox } from "@vscode/webview-ui-toolkit/react"
+import { Wrench } from "lucide-react"
+import { HTMLAttributes, useMemo } from "react"
+
+import { useAppTranslation } from "@/i18n/TranslationContext"
+
+import { Section } from "./Section"
+import { SectionHeader } from "./SectionHeader"
+import { SetCachedStateField } from "./types"
+
+type ToolSettingsProps = HTMLAttributes<HTMLDivElement> & {
+	disabledTools?: string[]
+	setCachedStateField: SetCachedStateField<"disabledTools">
+}
+
+// Import the constants from shared/tools.ts
+// Tool display names mapping
+const TOOL_DISPLAY_NAMES: Record<string, string> = {
+	execute_command: "Run commands",
+	read_file: "Read files",
+	fetch_instructions: "Fetch instructions",
+	write_to_file: "Write files",
+	apply_diff: "Apply changes",
+	search_files: "Search files",
+	list_files: "List files",
+	list_code_definition_names: "List definitions",
+	browser_action: "Use a browser",
+	use_mcp_tool: "Use MCP tools",
+	access_mcp_resource: "Access MCP resources",
+	ask_followup_question: "Ask questions",
+	attempt_completion: "Complete tasks",
+	switch_mode: "Switch modes",
+	new_task: "Create new task",
+	insert_content: "Insert content",
+	search_and_replace: "Search and replace",
+	codebase_search: "Codebase search",
+	update_todo_list: "Update todo list",
+}
+
+// Tools that are always available and cannot be disabled
+const ALWAYS_AVAILABLE_TOOLS = [
+	"ask_followup_question",
+	"attempt_completion",
+	"switch_mode",
+	"new_task",
+	"update_todo_list",
+]
+
+// Tool groups configuration
+const TOOL_GROUPS: Record<string, { tools: readonly string[] }> = {
+	read: {
+		tools: [
+			"read_file",
+			"fetch_instructions",
+			"search_files",
+			"list_files",
+			"list_code_definition_names",
+			"codebase_search",
+		],
+	},
+	edit: {
+		tools: ["apply_diff", "write_to_file", "insert_content", "search_and_replace"],
+	},
+	browser: {
+		tools: ["browser_action"],
+	},
+	command: {
+		tools: ["execute_command"],
+	},
+	mcp: {
+		tools: ["use_mcp_tool", "access_mcp_resource"],
+	},
+}
```

Comment:
test must validate that all tools within packages/types/src/tool.ts are represented in TOOL_GROUPS

Reply:
always explain which Roo mode was used to apply and implement and orchestrate the changes